extensions/java-joor-dsl/deployment/src/main/java/org/apache/camel/quarkus/dsl/java/joor/deployment/JavaJoorDslProcessor.java

```diff
@@ -0,0 +1,181 @@
+ */
+
+package org.apache.camel.quarkus.dsl.java.joor.deployment;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.annotations.ExecutionTime;
+import io.quarkus.deployment.annotations.Record;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.deployment.builditem.GeneratedClassBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
+import io.quarkus.deployment.pkg.builditem.CurateOutcomeBuildItem;
+import io.quarkus.deployment.pkg.steps.NativeBuild;
+import io.quarkus.maven.dependency.ResolvedDependency;
+import io.quarkus.paths.PathCollection;
+import io.quarkus.runtime.RuntimeValue;
+import org.apache.camel.CamelContext;
+import org.apache.camel.dsl.java.joor.CompilationUnit;
+import org.apache.camel.dsl.java.joor.MultiCompile;
+import org.apache.camel.quarkus.core.deployment.main.CamelMainHelper;
+import org.apache.camel.quarkus.core.deployment.spi.CamelContextBuildItem;
+import org.apache.camel.quarkus.dsl.java.joor.runtime.JavaJoorDslRecorder;
+import org.apache.camel.spi.Resource;
+import org.apache.camel.support.ResourceHelper;
+import org.apache.camel.util.FileUtil;
+import org.apache.camel.util.IOHelper;
+import org.apache.camel.util.StringHelper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class JavaJoorDslProcessor {
+
+    private static final Logger LOG = LoggerFactory.getLogger(JavaJoorDslProcessor.class);
+    private static final String FEATURE = "camel-java-joor-dsl";
+
+    private static final Pattern PACKAGE_PATTERN = Pattern.compile(
+            "^\\s*package\\s+([a-zA-Z][\\.\\w]*)\\s*;.*$", Pattern.MULTILINE);
+
+    @BuildStep
+    FeatureBuildItem feature() {
+        return new FeatureBuildItem(FEATURE);
+    }
+
+    @BuildStep(onlyIf = NativeBuild.class)
+    void loadClassesAOT(BuildProducer<JavaJoorGeneratedClassBuildItem> generatedClass) throws Exception {
+        LOG.debug("Loading .class resources");
+        CamelMainHelper.forEachMatchingResource(
+                resource -> {
+                    if (!resource.getLocation().endsWith(".class")) {
+                        return;
+                    }
+                    try (InputStream is = resource.getInputStream()) {
+                        generatedClass
+                                .produce(new JavaJoorGeneratedClassBuildItem(asClassName(resource), resource.getLocation(),
+                                        is.readAllBytes()));
+                    } catch (IOException e) {
+                        throw new RuntimeException(e);
+                    }
+                });
+    }
```

Comment:
Why would anybody want to pass .class files as resources? Is this also supported by plain Camel?

Reply:
Isn't Java actually loading those classes without our extension needing to do anything?

Reply:
> Why would anybody want to pass .class files as resources? Is this also supported by plain Camel?

Yes it also supported by Camel as you can see [here](https://github.com/apache/camel/blob/main/dsl/camel-java-joor-dsl/src/main/java/org/apache/camel/dsl/java/joor/ClassRoutesBuilderLoader.java)

Reply:
Hm... https://issues.apache.org/jira/browse/CAMEL-18014 says 

> When compiling with camel-java-dsl we can have it capture the compiled byte code, which allows us to write to disk, which we can make possible as a build step, that allows to load the routes from disk and avoid the compilation step at runtime.

@davsclaus is that use case described somewhere in more details? I mean what is build time and runtime for Camel JBang? I mostly wonder why would the build time part need to pass the .class files as resources to the runtime? Why can runtime simply pick those via stock java class path? Does this use case make any sense on Quarkus?

Reply:
I guess, I can remove the support of class files to keep it simple

Reply:
class files support removed